### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by david-waltermire-nist**
# Committer Notes

Code adjustments to resolve Java compile and PMD warnings.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/oss-maven/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/oss-maven/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- ~Have you written new tests for your core changes, as applicable?~
- ~Have you included examples of how to use your new feature(s)?~
- [x] Have you updated all website and readme documentation affected by the changes you made? (NOTE: website will be activated with Javadoc will be potentially activated by the merge of this PR.)

### CODE REVIEW

**Comment by aj-stein-nist** [comment]
Per tests that are currently failing and discussion with @david-waltermire-nist's work on releasing a 0.11.0 and 0.11.1 release, work with him on https://github.com/usnistgov/metaschema-java/pull/192 and continued integration on this PR.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -3,85 +3,80 @@ on:
...
         submodules: recursive
         fetch-depth: 0
     # -------------------------
-    # Java JDK 11
-    - name: Set up JDK 11
-      uses: actions/setup-java@5ffc13f4174014e2d4d4572b3d74c3fa61aeb2c2 # v3.3.0
+    # Java
+    # -------------------------
+    - name: Set up Maven
+      uses: stCarolas/setup-maven@07fbbe97d97ef44336b7382563d66743297e442f
```
Comment (aj-stein-nist) [inline .github/workflows/build.yml:33]: Concern: do we need to run a new setup-maven step for caching and that is why we need this new action and run it separately of setup-java?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -3,85 +3,80 @@ on:
...
         cache: 'maven'
+        server-id: ossrh # Value of the distributionManagement/repository/id field of the pom.xml
+        server-username: MAVEN_USERNAME # env variable for username in deploy
+        server-password: MAVEN_CENTRAL_TOKEN # env variable for token in deploy
+        gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }} # Value of the GPG private key to import
+        gpg-passphrase: MAVEN_GPG_PASSPHRASE # env variable for GPG private key passphrase
     - name: Initialize CodeQL
       if: github.event_name == 'push'
-      uses: github/codeql-action/init@46ed16ded91731b2df79a2893d3aea8e9f03b5c4 # v2.1.12
+      uses: github/codeql-action/init@46ed16ded91731b2df79a2893d3aea8e9f03b5c4
```
Comment (aj-stein-nist) [inline .github/workflows/build.yml:57]: Non-blocking: thank you for deleting the obsolete comment I love removing the cruft.

File: src/main/java/gov/nist/secauto/oscal/lib/profile/resolver/selection/Import.java [hunk 3]
```diff
@@ -158,41 +158,48 @@ private static void generateMetadata(
...
-        Version resolvedCatalogVersion
-            = VersionUtil.parseVersion(resolvedMetadata.getOscalVersion(), null, null);
+      Version resolvedCatalogVersion
+          = VersionUtil.parseVersion(resolved.getOscalVersion(), null, null);
 
-        if (importOscalVersion.compareTo(resolvedCatalogVersion) > 0) {
-          resolvedMetadata.setOscalVersion(importOscalVersion.toString());
-        }
+      if (importOscalVersion.compareTo(resolvedCatalogVersion) > 0) {
+        resolved.setOscalVersion(importOscalVersion.toString());
```
Comment (aj-stein-nist) [inline src/main/java/gov/nist/secauto/oscal/lib/profile/resolver/selection/Import.java:177]: Minor, non-blocking: should we be concerned about that `oscal-version` is a [string datatype](https://pages.nist.gov/OSCAL/reference/latest/profile/xml-reference/#/profile/metadata/oscal-version) and that comparison on non-semverish strings could throw up stack traces here? Minor but need to investigate further about what happens with setting and comparing `apple` and `banana` (not that I would do that like the funny, zany developer I am but the schema allows it).

**Review by aj-stein-nist — APPROVED** [review]
Minor comments and questions, but looks good thus far.

File: src/main/java/gov/nist/secauto/oscal/lib/profile/resolver/selection/Import.java (hunk 3 above)
Comment (david-waltermire-nist) [reply src/main/java/gov/nist/secauto/oscal/lib/profile/resolver/selection/Import.java:177]: Sure. I'd be happy to have better logic for this. Maybe create an issue to investigate?

File: .github/workflows/build.yml (hunk 1 above)
Comment (david-waltermire-nist) [reply .github/workflows/build.yml:33]: This paves the way for that. It also stabilizes the build environment on a specific version of Maven, which is important for reproducible builds.

File: src/main/java/gov/nist/secauto/oscal/lib/profile/resolver/selection/Import.java (hunk 3 above)
Comment (aj-stein-nist) [reply src/main/java/gov/nist/secauto/oscal/lib/profile/resolver/selection/Import.java:177]: https://github.com/usnistgov/liboscal-java/issues/158

**Pull request merged**